note_taker.bash

```diff
@@ -57,7 +57,65 @@ handle_options () {
             fi
             echo ""
             ;;
-        *)
+	"-a")
+	    # -a for all: pipes the whole notes file through less for quick, easy viewing
+	    cat "$NOTE_PAD_PATH" | less
+	    ;;
+	"-c")
+	    # -c for 'clear notes'
+	    # confirm user intention
+	    read -p "Clear all notes [y/n]? " -n 1 -r CLEAR_REPLY
+	    echo ""
+            # exit if not confirmed
+	    if [[ ! "$CLEAR_REPLY" =~ ^[Yy]$ ]]; then
+	    	echo "Cancelled."
+	    else 
+            	# clear notes file by overwriting it with blankness:
+		> "$NOTE_PAD_PATH"
+		# report success
+		echo "Cleared."
+	    fi
+            echo ""
+	    ;;
+	"-u")
+	    # -u for 'undo': deletes the most recent note. Specifically, it deletes lines starting
+            # with the next-to-last line and moving upwards through the file, until the next blank
+            # line it encounters.
+	    # confirm user intention
+	    read -p "Clear most recent note [y/n]? " -n 1 -r CLEAR_REPLY
+	    echo ""
+            # exit if not confirmed
+	    if [[ ! "$CLEAR_REPLY" =~ ^[Yy]$ ]]; then
+	    	echo "Cancelled."
+	    else 
+        	# get the total number of lines in the notes file
+		LINES=`cat "$NOTE_PAD_PATH" | wc -l`
+		# get the penultimate line, as the last line is always blank
+		((LINES--))
+		# get the text of the penultimate line and assign it to LINE_TEXT
+		LINE_NUMBER=$LINES
+		# get the text of the penultimate line and assign it to LINE_TEXT
+		LINE_TEXT=`tail -n +"$LINE_NUMBER" "$NOTE_PAD_PATH" | head -n 1`
+		# as long as LINE_TEXT is not empty and LINE_NUMBER is not 0
+		until [ -z "$LINE_TEXT" ] || [ "$LINE_NUMBER" -eq 0 ]; do
+			# delete the last line of the notes file by copying a shortened version
+			# to a temp file and then overwriting the original with the temp
+			head -n -1 "$NOTE_PAD_PATH" > temp.txt
```

Comment:
the bug is here:
on mac terminal bash, this causes:
```
head: illegal line count -- -1
```
I'll merge this and make the change.
```
cat "$NOTE_PAD_PATH" | tail -r | tail -n +2 | tail -r > temp.txt
```

Reply:
Nice fix!